x-pack/metricbeat/module/meraki/device_health/device_appliance_uplink_status_and_ha.go

```diff
@@ -0,0 +1,75 @@
+package device_health
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	meraki_api "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+func reportApplianceUplinkStatuses(reporter mb.ReporterV2, organizationID string, devices map[Serial]*Device, responseApplianceUplinkStatuses *meraki_api.ResponseApplianceGetOrganizationApplianceUplinkStatuses, lossLatencyUplinks []*Uplink) {
+
+	metrics := []mapstr.M{}
+
+	for _, uplink := range *responseApplianceUplinkStatuses {
+
+		if device, ok := devices[Serial(uplink.Serial)]; ok {
+			metric := mapstr.M{
+				"uplink.high_availablity.enabled": uplink.HighAvailability.Enabled,
+				"uplink.high_availablity.role":    uplink.HighAvailability.Role,
+				"uplink.last_reported_at":         uplink.LastReportedAt,
+				"device.address":                  device.Address,
+				"device.firmware":                 device.Firmware,
+				"device.imei":                     device.Imei,
+				"device.lan_ip":                   device.LanIP,
+				"device.location":                 device.Location,
+				"device.mac":                      device.Mac,
+				"device.model":                    device.Model,
+				"device.name":                     device.Name,
+				"device.network_id":               device.NetworkID,
+				"device.notes":                    device.Notes,
+				"device.product_type":             device.ProductType,
+				"device.serial":                   device.Serial,
+				"device.tags":                     device.Tags,
+			}
+
+			for k, v := range device.Details {
+				metric[fmt.Sprintf("device.details.%s", k)] = v
+			}
+
+			for _, item := range *uplink.Uplinks {
+
+				for _, lossLatencyMetric := range lossLatencyUplinks {
+					if lossLatencyMetric.Interface == item.Interface && string(lossLatencyMetric.DeviceSerial) == device.Serial && lossLatencyMetric.NetworkID == device.NetworkID {
+						for _, lossLatency := range lossLatencyMetric.Metrics {
+							//It seems there is bug in the client.Organizations.GetOrganizationDevicesUplinksLossAndLatency code returning differnt IP
+							//To mitigate, I am additionally printing the ip as seperate value, IMO it is odd these do not match.
+							// client.Appliance.GetOrganizationApplianceUplinkStatuses
+							metric["uplink.loss_latancy.ip"] = lossLatencyMetric.IP
+							metric["uplink.loss_latancy.@timestamp"] = lossLatency.Timestamp
```

Comment:
this timestamp represents the metric timestamp, and should be moved to the top level` @timestamp` field for this metric document. if we don't do this we end up with two `@timestamp`s in the document, one which is the time the event was reported and this one, which is the actual metric timestamp.

there's code in `ReportMetricsForOrganization` that pulls it out of the doc and sets it on the event.

Reply:
Ok, I made the change and fixed it, I moved this up to top level and the code in ReportMetricsForOrganization is correcting it (as before).